app/code/community/Bolt/Boltpay/Block/Checkout/Boltpay.php

```diff
@@ -222,6 +222,186 @@ public function getCartDataJs($checkoutType = self::CHECKOUT_TYPE_MULTI_PAGE)
         }
     }
 
+    /**
+     * Initiates the Bolt order creation / token receiving and sets up BoltCheckout with generated data.
+     * In BoltCheckout.configure success callback the order is saved in additional ajax call to
+     * Bolt_Boltpay_OrderController save action.
+     *
+     * @param string $checkoutType  'multi-page' | 'one-page' | 'admin' | 'firecheckout'
+     * @return string               BoltCheckout javascript
+     */
+    public function getCartDataJsForProductPage($checkoutType = self::CHECKOUT_TYPE_MULTI_PAGE)
+    {
+        try {
+            /* @var Mage_Sales_Model_Quote $sessionQuote */
+            $sessionQuote = $this->getSessionQuote($checkoutType);
+
+            /* @var Bolt_Boltpay_Helper_Api $boltHelper */
+            $boltHelper = Mage::helper('boltpay/api');
+
+            /** @var Bolt_Boltpay_Model_ShippingAndTax $shippingAndTaxModel */
+            $shippingAndTaxModel = Mage::getModel('boltpay/shippingAndTax');
+
+            $hintData = $this->getAddressHints($sessionQuote, $checkoutType);
+
+            $orderCreationResponse = json_decode('{"token" : "", "error": "'.Mage::helper('boltpay')->__('Unexpected error.  Please contact support for assistance.').'"}');
```

Comment:
note for product page checkout you don't need to make any API call. you just need to initialize JS object with product price etc.